rpc/src/v1/impls/mod.rs

```diff
@@ -55,22 +59,14 @@ fn dispatch_transaction<C, M>(client: &C, miner: &M, signed_transaction: SignedT
 	where C: BlockChainClient, M: MinerService {
 	let hash = signed_transaction.hash();
 
-	let import = {
-		miner.import_own_transaction(client, signed_transaction, |a: &Address| {
-			AccountDetails {
-				nonce: client.latest_nonce(&a),
-				balance: client.latest_balance(&a),
-			}
-		})
-	};
-
-	match import {
-		Ok(_) => to_value(&hash),
-		Err(e) => {
-			warn!("Error sending transaction: {:?}", e);
-			to_value(&H256::zero())
+	let import = miner.import_own_transaction(client, signed_transaction, |a: &Address| {
+		AccountDetails {
+			nonce: client.latest_nonce(&a),
+			balance: client.latest_balance(&a),
 		}
-	}
+	});
+
+	to_value(&import.map(|_| hash).unwrap_or(H256::zero()))
```

Comment:
will re-add, thanks for pointing it out. it might be more at home in the miner implementation itself, though.